ac/ac-imgClass/src/index.js

```diff
@@ -61,17 +61,21 @@ const dataStructure = {};
 // receives incoming data, and merges it with the reactive data using dataFn.*
 const mergeFunction = (object, dataFn) => {
   dataFn.objInsert(0, 'index');
-  const dataObj: Object = Array.isArray(object.data) ? {} : object.data;
-  const dataImgs = Object.keys(dataObj).filter(
-    x => dataObj[x].url !== undefined
-  );
-  if (dataObj !== {})
-    dataImgs.forEach((x, i) =>
-      dataFn.objInsert({ url: dataObj[x].url, category: '' }, i + 1)
-    );
   if (object.config.images)
     object.config.images.forEach((x, i) =>
-      dataFn.objInsert({ url: x, category: '' }, dataImgs.length + i + 1)
+      dataFn.objInsert({ url: x, category: '' }, i)
+    );
+
+  if (object.data === null || Array.isArray(object.data)) return;
+  const dataImgs = Object.keys(object.data).filter(
+    x => object.data[x].url !== undefined
+  );
+  if (object.data !== {})
+    dataImgs.forEach((x, i) =>
+      dataFn.objInsert(
+        { url: object.data[x].url, category: '' },
+        object.config.images.length + i
```

Comment:
Testing with images coming from and imgView activity as a product, but no image from the config. This line crashes as `object.config.images` is undefined. 

```
I20170829-11:31:37.872(2)? Exception in callback of async function: TypeError: Cannot read property 'length' of undefined
I20170829-11:31:37.872(2)?     at /home/louisfaucon/FROG/ac/ac-imgClass/dist/index.js:73:92
I20170829-11:31:37.872(2)?     at Array.forEach (native)
I20170829-11:31:37.872(2)?     at mergeFunction (/home/louisfaucon/FROG/ac/ac-imgClass/dist/index.js:72:36)
I20170829-11:31:37.873(2)?     at server/mergeData.js:55:13
I20170829-11:31:37.873(2)?     at runWithEnvironment (packages/meteor.js:1189:24)
```

Reply:
Please, fix the bug and write examples of jest tests for the merge function